src/org/thoughtcrime/securesms/ConversationItem.java

```diff
@@ -503,17 +503,13 @@ private void setContactPhoto() {
 
   private SpannableString linkifyMessageBody(SpannableString messageBody, boolean shouldLinkifyAllLinks) {
     if (!shouldLinkifyAllLinks) {
-	return messageBody;
+      return messageBody;
     }
 
-    boolean hasLinks = false;
-
     Pattern cmdPattern = Pattern.compile("(?<=^|\\s)/[a-zA-Z][a-zA-Z@\\d_/.-]{0,254}");
-    hasLinks = Linkify.addLinks(messageBody, cmdPattern, "cmd:", null, null) || hasLinks;
+    boolean hasLinks = Linkify.addLinks(messageBody, Linkify.EMAIL_ADDRESSES|Linkify.WEB_URLS|Linkify.PHONE_NUMBERS);
 
-    hasLinks = Linkify.addLinks(messageBody, Patterns.EMAIL_ADDRESS, "mailto:", null, null) || hasLinks;
-    hasLinks = Linkify.addLinks(messageBody, Patterns.WEB_URL, "http://", new String[]{"http://", "https://"},  null, null) || hasLinks;
```

Comment:
This method can't be used in Android M (6) and older. (note the extra `new String[]{"http://", "https://"}` array argument; you can't do this on older devices).

I now just went back to the old approach of `Linkify.addLinks(messageBody, Linkify.EMAIL_ADDRESSES|Linkify.WEB_URLS|Linkify.PHONE_NUMBERS);`, does this have any drawback?

Reply:
From @adbenitez 
> this has drawbacks, the linkify is lost if this come after the command linkify and if you put it before then numbers in commands are linked as phone numbers etc